### ISSUE TITLE
required namespace positional argument missing error in walkthrough

### ISSUE THREAD

**Issue by ianmiell** [zalando-incubator/kopf#79]
Got to this part of the walkthrough
https://kopf.readthedocs.io/en/stable/walkthrough/creation/

and saw this error in the `kopf ephemeral.py --verbose` output:

```
[2019-05-27 18:33:56,232] kopf.reactor.handlin [DEBUG   ] [myproject/my-claim] Creation event: {'apiVersion': 'zalando.org/v1', 'kind': 'EphemeralVolumeClaim', 'kopf': {'dummy': '2019-05-27T17:33:31.599067'}, 'metadata': {'annotations': {'kubectl.kubernetes.io/last-applied-configuration': '{"apiVersion":"zalando.org/v1","kind":"EphemeralVolumeClaim","metadata":{"annotations":{},"name":"my-claim","namespace":"myproject"},"spec":{"size":"10G"}}\n'}, 'creationTimestamp': '2019-05-27T17:29:58Z', 'finalizers': ['KopfFinalizerMarker'], 'generation': 1, 'name': 'my-claim', 'namespace': 'myproject', 'resourceVersion': '776587', 'selfLink': '/apis/zalando.org/v1/namespaces/myproject/ephemeralvolumeclaims/my-claim', 'uid': '0cbcaf4b-80a5-11e9-a261-46c1de1eb0ee'}, 'spec': {'size': '10G'}, 'status': {'kopf': {'progress': {'create_fn': {'delayed': '2019-05-27T17:33:56.117633', 'retries': 4, 'started': '2019-05-27T17:29:55.703835'}}}}}
[2019-05-27 18:33:56,233] kopf.reactor.handlin [DEBUG   ] [myproject/my-claim] Invoking handler 'create_fn'.
[2019-05-27 18:33:56,233] kopf.reactor.handlin [ERROR   ] [myproject/my-claim] Handler 'create_fn' failed with an exception. Will retry.
Traceback (most recent call last):
  File "/Users/imiell/anaconda3/lib/python3.7/site-packages/kopf/reactor/handling.py", line 382, in _execute
    lifecycle=lifecycle,  # just a default for the sub-handlers, not used directly.
  File "/Users/imiell/anaconda3/lib/python3.7/site-packages/kopf/reactor/handling.py", line 478, in _call_handler
    **kwargs,
  File "/Users/imiell/anaconda3/lib/python3.7/site-packages/kopf/reactor/invocation.py", line 70, in invoke
    result = task.result()  # re-raises
  File "/Users/imiell/anaconda3/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
TypeError: create_fn() missing 1 required positional argument: 'namespace'
```

I was in a 'myproject' namespace as system:admin in a minishift 3.11 (sic) cluster - are there problems with running this against 1.11 k8s APIs?

**Comment by nolar** [comment]
@ianmiell Thanks for reporting. 

Indeed, the EphemeralVolumeClaim operator was not tested to work — as it is not finished yet (the critical missing feature is already implemented, but not merged yet — will be released in couple days).

The issue with the kwarg is fixed in #80.

Meanwhile, as a quick work-around, you can replace `namespace` kwarg with `meta`, and use `meta['namespace']` to refer to the namespace. Or, similarly, `body` & `body['metadata']['namespace']`.

**Comment by ianmiell** [comment]
Thanks @nolar. This looks like a great project, let me know if I can help. I have a little experience in automated testing of k8s clusters.

**Comment by nolar** [comment]
Fixed with `kopf==0.12`: https://github.com/zalando-incubator/kopf/releases/tag/0.12
